src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+            f"main.csv"
+        )
+        return directory / filename
+
+    def load_rolling_sum_data(self, filter: bool = False) -> pd.DataFrame:
+        """Load rolling sum data"""
+        rollsum_fp = self._get_rolling_sum_filepath()
+        df = pd.read_csv(rollsum_fp, parse_dates=["T"])
+        if filter:
+            t_col = df.columns[0]
+            df = df[
+                (df[t_col].dt.date >= self.date_min)
+                & (df[t_col].dt.date <= self.date_max)
+            ]
+        return df
+
+    def identify_dry_spells(self, reprocess: bool = False) -> pd.DataFrame:
+        """
+        Identifies dry spells based on latest rolling
+        sum values.
+
+        :param reprocess: Boolean, if `True` reprocesses
+        aggregated data by calculating rolling sums.
+        """
+        if reprocess:
+            self.calculate_rolling_sum()
+
+        rollsum_fp = self._get_rolling_sum_filepath()
+        df = pd.read_csv(rollsum_fp, parse_dates=["T"])
+
+        t_col = df.columns[0]
+        adm_col = df.columns[1]
+        rs_col = df.columns[2]
+
+        # Identify all dry spells and unique consecutive groups
+        df["ds"] = df[rs_col] <= self.rainfall_mm
+        df["dsg"] = (df.groupby(adm_col)["ds"].diff() != 0).cumsum()
+
+        # Generate data frame of dry spells
+        ds_df = (
+            df[df["ds"]]
+            .groupby("dsg")
+            .agg(
+                adm_col=(adm_col, "unique"),
+                ds_confirmation=(t_col, "min"),
+                ds_last_date=(t_col, "max"),
+            )
+            .reset_index(drop=True)
+            .assign(
+                adm_col=lambda x: x.adm_col.str[0],
+                ds_first_date=lambda x: x.ds_confirmation
+                - pd.to_timedelta(self.rolling_window - 1, unit="d"),
+                ds_duration=lambda x: (
+                    x.ds_last_date - x.ds_first_date
+                ).dt.days
+                + 1,
+            )
+            .rename(columns={"adm_col": adm_col})
+        )
+
+        # Calculate rainfall within each dry spell
+        precip_df = self.load_aggregated_data()
+        precip_col = precip_df.columns[1]
+
+        total_df = pd.merge(
+            left=ds_df, right=precip_df, on=adm_col, how="outer"
+        )
+
+        # Filter dates to just those within the dry spell and sum
+        total_df = total_df[total_df[t_col] >= total_df["ds_first_date"]]
+        total_df = total_df[total_df[t_col] <= total_df["ds_last_date"]]
+        total_df = (
+            total_df.groupby([adm_col, "ds_confirmation"])[precip_col]
+            .agg("sum")
+            .reset_index()
+        )
+
+        # Add dry spell rainfall to main data frame
+        total_df.rename(columns={precip_col: "ds_rainfall"}, inplace=True)
+
+        ds_df = pd.merge(
+            left=ds_df,
+            right=total_df,
+            on=[adm_col, "ds_confirmation"],
+            how="left",
+        )
```

Comment:
I was trying to comeup with an easier method to compute the total rainfall but couldn't so this looks good!